lib/eventq/eventq_aws/aws_eventq_client.rb

```diff
@@ -15,25 +15,37 @@ def initialize(options)
         @serialization_manager = EventQ::SerializationProviders::Manager.new
         @signature_manager = EventQ::SignatureProviders::Manager.new
 
-        # this array is used to record known event types
-        @known_event_types = []
-
+        # this hash is used to record known event types:
+        # key = event_type / name
+        # value = topic arn
+        @known_event_types = {}
       end
 
+      # Returns true if the event has already been registerd, or false
+      # otherwise.
+      #
+      # @param [String] event_type
+      # @param [String] region
+      #
+      # @return [Boolean]
       def registered?(event_type, region = nil)
         topic_key = "#{region}:#{event_type}"
-        @known_event_types.include?(topic_key)
+        @known_event_types.key?(topic_key)
       end
 
+      # Registers the event event_type and returns its topic arn.
+      #
+      # @param [String] event_type
+      # @param [String] region
+      #
+      # @return [String]
```

Comment:
Since this changes the method signature this ought to be a breaking change / major version bump - i.e. tag it as v3.x.x